dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/deaccession/DatasetDeaccessionServiceIT.java

```diff
@@ -0,0 +1,97 @@
+package edu.harvard.iq.dataverse.dataset.deaccession;
+
+import edu.harvard.iq.dataverse.DatasetServiceBean;
+import edu.harvard.iq.dataverse.DatasetVersionServiceBean;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+import javax.inject.Inject;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class DatasetDeaccessionServiceIT extends WebappArquillianDeployment {
+
+    @EJB
+    private DatasetDeaccessionService deaccessionService;
+
+    @Inject
+    private DatasetServiceBean datasetService;
+
+    @Inject
+    private DataverseSession dataverseSession;
+
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+
+    @Inject
+    private DatasetVersionServiceBean versionsService;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void shouldDeaccessVersion() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+        DatasetVersion versionToBeDeaccessed = dataset.getReleasedVersion();
+
+        // when
+        deaccessionService.deaccessVersion(versionToBeDeaccessed, "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+
+        assertEquals(DatasetVersion.VersionState.DEACCESSIONED, versionsService.find(versionToBeDeaccessed.getId()).getVersionState());
+        assertEquals(dataset.getVersions().size(), dbDataset.getVersions().size());
```

Comment:
`dataset.getVersions().size()` should be assigned to int before passing it to service otherwise the value will be mutated.